### PR TITLE
Adding Scouting UI / Functionality to Living Room + Yard

### PR DESCRIPTION
**Pull request by imann24**
**Prefabs**
- Made Scouting UI its own prefab
- Separated scouting UI out from living room + yard prefabs
- Adjusted UI spacing on living room + yard prefabs
- Sorted prefabs into sub folder
- Made RedeemDisplay into its own prefab 

**Scenes**
- Hooked up refs for scouting UI and added dog browser to Living Room + Yard

**Scripts**
- Added functionality to Redeem gifts from dogs
- Began writing (and then disabled) functionality for resend
- Hooked up RedeemDisplay to scouting system

### CODE REVIEW

**Comment by gracebarsny** [comment]
MCs in the LivingRoom and Yard scenes.

**Comment by gracebarsny** [comment]
It doesn't look like you combined the scouting and gifts code. Seems like you threw out James's work. The gifts shouldn't be auto-redeemed when the dog returns. It should look the way it did before, with the gift on the dog slot that the user can tap to redeem, which will trigger a popup.

I'd also add that the DogOutsideSlots shouldn't be interactable when there's already a dog in them.

File: Assets/Scripts/Controller/PickupPup/PPGameController.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public class PPGameController : GameController, ICurrencySystem
 {
 	#region Static Accessors
 
-	// Returns the Instance cast to the sublcass
+	// Returns the Instance cast to the sublcassmonob
```
Comment (gracebarsny) [inline Assets/Scripts/Controller/PickupPup/PPGameController.cs]: `sublcassmonob`
The what now?

File: Assets/Scripts/Controller/PickupPup/PPGameController.cs [hunk 2]
```diff
@@ -291,20 +289,22 @@ public bool TryTakeCurrency(CurrencyData currency)
...
 		CurrencyData data = giftController.GetGiftFromDog(dog);
-		dataController.ChangeCurrencyAmount(data.Type, data.Amount);
 		return data;
 	}
 
     public bool TryBuyItem(int value, CurrencyType valueCurrencyType,
         int cost, CurrencyType costCurrencyType)
     {
-        if (!CanAfford(costCurrencyType, cost))
+        if (CanAfford(costCurrencyType, cost))
```
Comment (gracebarsny) [inline Assets/Scripts/Controller/PickupPup/PPGameController.cs]: Remove space between if and (

**Comment by gracebarsny** [comment]
Nowhere in your commit notes do you mention you sorted all the prefabs into subfolders. Will cause many MCs -- VERY risky to do this right before a major deadline. Please consult me next time.

**Comment by imann24** [comment]
@gracebarsny I've rearranged previously and not encountered issues. Simply merging in command line has fixed without any real MC's every time.

File: Assets/Scripts/Controller/PickupPup/PPGameController.cs (hunk 1 above)
Comment (imann24) [inline Assets/Scripts/Controller/PickupPup/PPGameController.cs]: Good question

**Comment by gracebarsny** [comment]
@imann24 That's a relief. Still let me know or at least make note of it!

**Comment by imann24** [comment]
@gracebarsny Copy that

**Comment by imann24** [comment]
@gracebarsny Ready for full review. Removed the "Resend" feature as per Sam's notes

**Comment by imann24** [comment]
@gracebarsny SFX notes may be related to future PR. Will hold off on addressing that note until all PR's are merged. Would you mind making a separate task / subtask for that?

**Comment by imann24** [comment]
@gracebarsny Could not repro this bug "I get a MissingReferenceException on in DogOutsideSlot line 169 when a dog's timer gets to roughly 17 seconds on the first go-around. Unity says you're trying to access a GameObject that's been destroyed. Doesn't seem to interrupt the game." Please verify you're still experiencing it, and if so, can you provide more detailed steps to observe?

**Comment by gracebarsny** [comment]
Most issues from my last review are totally resolved. I couldn't reproduce that MissingReferenceException either.

Primary issue was that dogs would disappear from the slot on reload. Details below.
- This would occur both while the timer was running and when the gift was waiting to be redeemed.
- Would only occur on the first reload. If the dog was there, it would stay there for future reloads. If the dog disappeared, it never returned.
- Could happen in all 4 slots
- Saw it happen for all timers > 1 second but did not test the 50-minute timer
- Worked more often than not, but the bug was frequent enough to be a problem
- Not dog-specific. I could resend the same dog and it'd be fine.

Found out you could send the same dog out while it was already scouting. The timers would sync up to the timer that was first set.

Got a NullReferenceExceptions in RedeemDisplay line 72 on the first dog I tested (2-minute timer). But couldn't reproduce it.

**Pull request merged**